### PR TITLE
BN-1138/113434 group and series asset token

### PR DESCRIPTION
**Pull request by nandotorterolo**
## Purpose

TAMv2 Assets
A TAMv2 asset is a token that has the following attributes:

- A group identifier
- A series identifier
- A quantity

Asset Lifecycle

- **Mint a group constructor token**
- Mint a series constructor token (this can be done in the same transaction as the group constructor is minted)
- Mint an asset token using the group and series constructor token just minted.

### Note

This PR is linked to the following PRs, and only group constructor token, with empty series was implemented.

- https://github.com/Topl/protobuf-specs/pull/78
- https://github.com/Topl/BramblSc/pull/86
- https://github.com/Topl/Bifrost/pull/3006

## Approach
-  Mint a group constructor token
- test a group constructor token with success result 
- test a group constructor token duplicated, with a previuos iotx,   with invalid result
- test a group constructor token duplicated, in the same iotx, with invalid result 
- test a group constructor which a reference an UTXO which not contain TOPLs
 
## Testing
- preparePR, integration test

## Tickets
https://topl.atlassian.net/browse/BN-909

### CODE REVIEW

File: blockchain/src/main/scala/co/topl/blockchain/DataStores.scala [hunk 1]
```diff
@@ -28,7 +29,8 @@ case class DataStores[F[_]](
   registrations:           Store[F, StakingAddress, ActiveStaker],
   blockHeightTree:         Store[F, Long, BlockId],
   epochData:               Store[F, Epoch, EpochData],
-  registrationAccumulator: Store[F, StakingAddress, Unit]
+  registrationAccumulator: Store[F, StakingAddress, Unit],
+  groups:                  Store[F, GroupId, Value.Group]
```
[the remark this answers is not in the document]
Comment (SeanCheatham) [reply blockchain/src/main/scala/co/topl/blockchain/DataStores.scala:33]: I think you only need to check that within a transaction, there aren't two "Group" outputs that try to reference the same LVL input, but that can be performed at a syntax level.

File: ledger/src/main/scala/co/topl/ledger/interpreters/TransactionSemanticValidation.scala [hunk 2]
```diff
@@ -111,4 +111,23 @@ object TransactionSemanticValidation {
       )
       .pure[F]
 
+  /**
+   * is this transaction a group constructor token, and the UTXO referenced contains LVLs, that are paid for minting it?
+   */
+  private def groupConstructorTokenLvlValidation[F[_]: Monad](
+    fetchTransaction: TransactionId => F[IoTransaction]
```
[the remark this answers is not in the document]
Comment (nandotorterolo) [reply ledger/src/main/scala/co/topl/ledger/interpreters/TransactionSemanticValidation.scala:118]: Thank you for the review, and the direction about how to proceed.

File: blockchain/src/main/scala/co/topl/blockchain/DataStores.scala (hunk 1 above)
[the remark this answers is not in the document]
Comment (nandotorterolo) [reply blockchain/src/main/scala/co/topl/blockchain/DataStores.scala:33]: Correct, thank you for the suggestion.

**Comment by nandotorterolo** [comment]
Closing this Pr, implementation is not correct. I will keep the branch, the test could be useful in the future.

**Pull request closed without merging**